### DISCUSSION TITLE
Transparency?

### DISCUSSION THREAD

**Discussion by asherevan** [russhughes/st7789_mpy#175] [General]
Does this library support drawing transparent images?

**Comment by russhughes** [comment]
Transparency is not supported due to the absence of a frame buffer. However, you can still achieve partial transparency drawing PNGs by providing a mask. A True value for the mask parameter will prevent pixels with zero alpha channel values from displaying.

Drawing masked PNGs is slower, as each visible line segment is drawn individually. Please take a look at the alien.py program in the examples/png folder for an example of using a mask.

If you are using a ESP32-S3 device the [s3lcd driver](https://github.com/russhughes/s3lcd) driver uses a frame buffer and supports transparency.

**Comment by asherevan** [comment]
What do you mean by partial?
